### PR TITLE
ll_latency_test: fix time calculation

### PR DESCRIPTION
**Pull request by wenkaidu**

### CODE REVIEW

File: tools/p2p-latency-test/ll_latency_test.cpp [hunk 1]
```diff
@@ -91,7 +91,8 @@ __global__ void PingKernel(LLFifoLine* local_flag, LLFifoLine* remote_flag, uint
     while (readLL(local_flag+tid, i) != i);
   }
   uint64_t end_time = wall_clock64();
-  *time_delta = end_time - start_time;
+  __syncthreads();
+  if (tid == 0) *time_delta = end_time - start_time;
```
Comment (nusislam) [inline tools/p2p-latency-test/ll_latency_test.cpp:95]: Instead of taking the latency of thread-0, should we take the max latency among all the threads?

Comment (wenkaidu) [reply tools/p2p-latency-test/ll_latency_test.cpp:95]: __syncthreads() will wait for all threads to reach the barrier, thus we will measure max latency

Comment (nusislam) [reply tools/p2p-latency-test/ll_latency_test.cpp:95]: The `end_time` is taken before calling `__syncthreads()`. The time required to sync among the threads is not being measured in the latency.

**Review by nusislam — APPROVED** [review]

File: tools/p2p-latency-test/ll_latency_test.cpp (hunk 1 above)
Comment (gilbertlee-amd) [reply tools/p2p-latency-test/ll_latency_test.cpp:95]: Only thread 0 needs to collect wall_clock64().
if (tid == 0) *time_delta = wall_clock64() - start_time;

Comment (wenkaidu) [reply tools/p2p-latency-test/ll_latency_test.cpp:95]: I know, but think not worth it to create diversion. Would rather let all thread do the same

Comment (gilbertlee-amd) [reply tools/p2p-latency-test/ll_latency_test.cpp:95]: You potentially add additional latency if other wavefronts (not thread 0's) all have to compute wall_clock64() before thread 0's wavefront.

Comment (wenkaidu) [reply tools/p2p-latency-test/ll_latency_test.cpp:96]: as you wish...

**Pull request merged**